### PR TITLE
CMake versioning and Docker build improvements

### PR DESCRIPTION
**Pull request by mkaring**
The following changes are contained in this PR:

1. The versioning for cmake is now provided by `git` and does no longer require a manual update
2. The github actions now create debian packages for both the release and the release with debug information builds. Tests are executed for both builds.
3. The server no longer runs as root user in the docker container, instead a new user called "illarion" is created and is executing the server.
4. The labeling for the docker container is fixed, so the image is properly linked to the repository

### CODE REVIEW

**Comment by vilarion** [comment]
When I bind the config file into the container, it cannot be read by the non-priviledged user.
On the host: `-rw-r----- 1 devserver      devserver      554 Nov 30 00:49 /etc/illarion/dev.conf`
Inside the container: `-rw-r----- 1 root root 554 Nov 29 23:49 /etc/illarion.conf`

File: setup/docker/base/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,19 @@ RUN \
...
+    useradd -rm -d /home/illarion --shell /bin/bash -g illarion -G users illarion && \
+    chown :illarion /usr/share/illarion/map && \
+    chown :illarion /usr/share/illarion/scripts && \
+    chmod -R a+r /scripts /maps && \
+    chmod -R g+w /usr/share/illarion/scripts /usr/share/illarion/map && \
+    chmod a+r /etc/illarion.conf && \
+    chmod a+rx /usr/share/illarion/pre-reload
+
+USER illarion
+WORKDIR /home/illarion
```
Comment (vilarion) [inline setup/docker/base/Dockerfile:33]: I think we can remove the custom user altogether. We can grant write access for everyone to `/usr/share/illarion/map` and `/usr/share/illarion/scripts` and then set the user via docker-compose to run as e.g. devserver.
In this scenario the executing user on the host would be responsible for running the container in a secure way, which sounds fine.

**Review by vilarion — CHANGES_REQUESTED** [review]

**Review by vilarion — CHANGES_REQUESTED** [review]
I think `/usr/share/illarion/map` needs to be in a named volume, for maps to persist between server updates.

File: setup/docker/base/Dockerfile (hunk 1 above)
Comment (mkaring) [reply setup/docker/base/Dockerfile]: Done that.

**Review by vilarion — APPROVED** [review]

**Pull request merged**